src/arrayMethodJoin.js

```diff
@@ -5,8 +5,23 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedString = '';
+
+    if (separator === undefined) {
+      separator = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        joinedString += this[i];
+      }
+
+      if (i !== this.length - 1) {
+        joinedString += separator;
```

Comment:
Другий варіант це тут додати умову якщо потрібна, приблизно так:
```suggestion
        joinedString += separator === undefined ? ',' : separator ;
```